x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleServiceTests.java

```diff
@@ -497,4 +503,96 @@ public void testParsingOriginationDateBeforeIndexCreation() {
+                .setPhase("warm")
+                .setAction("shrink")
+                .setStep(randomFrom(SetSingleNodeAllocateStep.NAME, CheckShrinkReadyStep.NAME,
+                    ShrinkStep.NAME, ShrunkShardsAllocatedStep.NAME))
+                .build().asMap())
+            .numberOfShards(randomIntBetween(1, 5)).numberOfReplicas(randomIntBetween(0, 5)).build();
+        ImmutableOpenMap.Builder<String, IndexMetadata> indices = ImmutableOpenMap.<String, IndexMetadata> builder()
+            .fPut("no_danger", nonDangerousIndex)
+            .fPut("danger", dangerousIndex);
+
+        Metadata metadata = Metadata.builder()
+            .putCustom(IndexLifecycleMetadata.TYPE, new IndexLifecycleMetadata(Collections.emptyMap(), OperationMode.RUNNING))
+            .indices(indices.build())
+            .persistentSettings(settings(Version.CURRENT).build())
+            .build();
+
+        state = ClusterState.builder(ClusterName.DEFAULT)
+            .metadata(metadata)
+            .nodes(DiscoveryNodes.builder().localNodeId(nodeId).masterNodeId(nodeId)
+                .add(masterNode)
+                .add(DiscoveryNode.createLocal(
+                    NodeRoles.masterNode(settings(Version.CURRENT).build()),
+                    new TransportAddress(TransportAddress.META_ADDRESS, 9301),
+                    "regular_node"))
+                .add(DiscoveryNode.createLocal(
+                    NodeRoles.masterNode(settings(Version.CURRENT).build()),
+                    new TransportAddress(TransportAddress.META_ADDRESS, 9302),
+                    "shutdown_node"))
+                .build())
+            .build();
+
+        // No danger yet, because no node is shutting down
+        assertThat(IndexLifecycleService.indicesOnShuttingDownNodesInDangerousStep(state, "regular_node"),
+            equalTo(Collections.emptySet()));
+        assertThat(IndexLifecycleService.indicesOnShuttingDownNodesInDangerousStep(state, "shutdown_node"),
+            equalTo(Collections.emptySet()));
+
+        state = ClusterState.builder(state)
+            .metadata(Metadata.builder(state.metadata())
+                .putCustom(NodesShutdownMetadata.TYPE, new NodesShutdownMetadata(Collections.singletonMap("shutdown_node",
+                    SingleNodeShutdownMetadata.builder()
+                        .setNodeId("shutdown_node")
+                        .setReason("shut down for test")
+                        .setStartedAtMillis(randomNonNegativeLong())
+                        .setType(SingleNodeShutdownMetadata.Type.RESTART)
+                        .build())))
+                .build())
+            .build();
+
+        assertThat(IndexLifecycleService.indicesOnShuttingDownNodesInDangerousStep(state, "regular_node"),
+            equalTo(Collections.emptySet()));
+        // No danger, because this is a "RESTART" type shutdown
+        assertThat(IndexLifecycleService.indicesOnShuttingDownNodesInDangerousStep(state, "shutdown_node"),
```

Comment:
Nitpick: I think the comment here should be promoted to a message on the assert, e.g.
```java
assertThat("RESTART type shutdowns should not be considered dangerous", IndexLifecycleService.indicesOnShuttingDownNodesInDangerousStep(state, "shutdown_node"),
            equalTo(Collections.emptySet()));
```